### ISSUE TITLE
Navigating back to previous page forgets scroll position after initial load on New York Times

### ISSUE THREAD

**Issue by yoasif** [mozilla-mobile/focus-android#2828]
When reading articles on nytimes.com, I have found that going back to the previous page takes me back to the top of the page instead of where I was before I went to a new page. 

### Steps to reproduce

1. Open https://mobile.nytimes.com/2018/06/29/briefing/supreme-court-capital-gazette-sicario.html?action=click&module=Briefings&pgtype=Homepage
2. Click on a linked article that is partway down the page - I used "shooting at a community newspaper chain" (https://www.nytimes.com/2018/06/28/us/capital-gazette-annapolis-shooting.html)
3. Wait for page to load
4. Touch "back" action in Android

### Expected behavior

The previous page should be scrolled to where I was when I clicked the "shooting at a community newspaper chain" link.

### Actual behavior

Initially, the page is scrolled to where I was when I clicked "shooting at a community newspaper chain". After a second or two, the page appears to continue to load and the browser jumps back to the top of the page. This doesn't happen in Fennec or in the OEM provided browser on this device. 

### Device information

* Android device: LG-H810, Android 5.1
* Focus version: 6.2 (Build #221791209 Emoji image of 'lizard' 62.0.20180625141512)

**Comment by pocmo** [comment]
I couldn't reproduce this (62.20180625100038; Pixel XL, Android 9). I opened nytimes.com, waited for it to load, scrolled down, clicked on an article, waited for it to load, clicked back.... and it loaded the previous page in the same scroll position as before.

**Comment by pocmo** [comment]
@npark-mozilla Can you test this too?

**Comment by yoasif** [comment]
@pocmo Does the page appear to continue loading after you click back? It does for me, and initially starts in the same scroll position, but jumps back up to the top of the page after a few seconds. 

Might be because my device is slower than yours?

**Comment by pocmo** [comment]
I tried again and it doesn't appear loading and doesn't jump back after waiting for a while.

**Comment by yoasif** [comment]
I hate to belabor the point here, but it works okay on nytimes.com -- did you try starting from 

https://mobile.nytimes.com/2018/06/29/briefing/supreme-court-capital-gazette-sicario.html?action=click&module=Briefings&pgtype=Homepage

and clicking a link from there?

**Comment by pocmo** [comment]
Yes, I did. Oh, and if I'm not able to reproduce then this doesn't mean we do not care! :) I'll wait what @npark-mozilla reports back - He'll be able to test more thoroughly. Also @snorp mentioned that this could be related to bfcache and that this may have been unreliable in Fennec too.

**Comment by yoasif** [comment]
Sorry, I was unsure whether my edited comment would produce an email notification on GitHub, so I just wanted to make sure we were on the same page (since my reading of your comment was that you tried nytimes.com). 

Apologies if I was mistaken on my initial reading, thanks for trying to reproduce!

**Comment by npark-mozilla** [comment]
I kinda reproduced this on my 3rd attempt, this seems to happen if I click a link to the story when the first page was still in process of loading (more likely to reproduce the issue when there are a lot of data to load). It will take me to the linked page, and when I press back button, it will initially go back to where I pressed the link, but then it'll readjust the screen.  I think we saw something like this before, but at the time I don't think we found a workaround for this.
